AWS update breaking `deploy-s3` task on Travis?

From Travis:
```
[0K$ npm run deploy-s3 -- -b cesium-dev -d cesium/$TRAVIS_BRANCH --confirm -c 'no-cache'

> cesium@1.48.0-additionalProjections28414 deploy-s3 /home/travis/build/AnalyticalGraphicsInc/cesium
> gulp deploy-s3 "-b" "cesium-dev" "-d" "cesium/additionalProjections" "--confirm" "-c" "no-cache"

[19:45:11] Using gulpfile ~/build/AnalyticalGraphicsInc/cesium/gulpfile.js
[19:45:11] Starting 'deploy-s3'...
[19:45:11] 'deploy-s3' errored after 18 ms
[19:45:11] TypeError: Cannot promisify an API that has normal methods with 'Async'-suffix

    See http://goo.gl/MqrFmX

    at checkValid (/home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/bluebird/js/release/promisify.js:54:27)
    at promisifiableMethods (/home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/bluebird/js/release/promisify.js:77:5)
    at promisifyAll (/home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/bluebird/js/release/promisify.js:241:9)
    at new Promise.promisifyAll (/home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/bluebird/js/release/promisify.js:311:12)
    at deployCesium (/home/travis/build/AnalyticalGraphicsInc/cesium/gulpfile.js:344:14)
    at Gulp.<anonymous> (/home/travis/build/AnalyticalGraphicsInc/cesium/gulpfile.js:315:9)
    at module.exports (/home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/orchestrator/index.js:134:8)
    at /home/travis/build/AnalyticalGraphicsInc/cesium/node_modules/gulp/bin/gulp.js:129:20
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
    at Function.Module.runMain (module.js:695:11)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! cesium@1.48.0-additionalProjections28414 deploy-s3: `gulp deploy-s3 "-b" "cesium-dev" "-d" "cesium/additionalProjections" "--confirm" "-c" "no-cache"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the cesium@1.48.0-additionalProjections28414 deploy-s3 script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2018-08-30T19_45_11_871Z-debug.log

```

The line referenced in here seems to call the `aws-sdk`, which was updated very recently.

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -438,8 +438,8 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
                         CacheControl : cacheControl
                     };
 
-                    return s3.putObject(params)
-                        .promise().then(function() {
+                    return s3.putObject(params).promise()
+                        .then(function() {
                             uploaded++;
                         })
                         .catch(function(error) {
@@ -521,13 +521,12 @@ function getMimeType(filename) {
 
 // get all files currently in bucket asynchronously
 function listAll(s3, bucketName, prefix, files, marker) {
-    return s3.listObjects.promise({
-        Bucket : bucketName,
-        MaxKeys : 1000,
-        Prefix : prefix,
-        Marker : marker
-    })
-    .then(function(data) {
+    return s3.listObjects({
+        Bucket: bucketName,
+        MaxKeys: 1000,
+        Prefix: prefix,
+        Marker: marker
+    }).promise().then(function(data) {
         var items = data.Contents;
         for (var i = 0; i < items.length; i++) {
             files.push(items[i].Key);

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -26,7 +26,7 @@ var Promise = require('bluebird');
 var requirejs = require('requirejs');
 var Karma = require('karma').Server;
 var yargs = require('yargs');
-var aws = require('aws-sdk');
+var S3 = require('aws-sdk/clients/s3');
 var mime = require('mime');
 var compressible = require('compressible');
 
@@ -338,15 +338,14 @@ gulp.task('deploy-s3', function(done) {
 function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
     var readFile = Promise.promisify(fs.readFile);
     var gzip = Promise.promisify(zlib.gzip);
-    var getCredentials = Promise.promisify(aws.config.getCredentials, {context: aws.config});
     var concurrencyLimit = 2000;
 
-    var s3 = new Promise.promisifyAll(new aws.S3({
+    var s3 = new S3({
         maxRetries : 10,
         retryDelayOptions : {
             base : 500
         }
-    }));
+    });
 
     var existingBlobs = [];
     var totalFiles = 0;
@@ -354,10 +353,8 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
     var skipped = 0;
     var errors = [];
 
-    return getCredentials()
-    .then(function() {
-        var prefix = uploadDirectory + '/';
-        return listAll(s3, bucketName, prefix, existingBlobs)
+    var prefix = uploadDirectory + '/';
+    return listAll(s3, bucketName, prefix, existingBlobs)
         .then(function() {
             return globby([
                 'Apps/**',
@@ -377,8 +374,7 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
             ], {
                 dot : true // include hidden files
             });
-        })
-        .then(function(files) {
+        }).then(function(files) {
             return Promise.map(files, function(file) {
                 var blobName = uploadDirectory + '/' + file;
                 var mimeLookup = getMimeType(blobName);
@@ -407,25 +403,24 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
                     existingBlobs.splice(index, 1);
 
                     // get file info
-                    return s3.headObjectAsync({
-                        Bucket : bucketName,
-                        Key : blobName
-                    })
-                    .then(function(data) {
-                        if (data.ETag !== ('"' + hash + '"') ||
-                            data.CacheControl !== cacheControl ||
-                            data.ContentType !== contentType ||
-                            data.ContentEncoding !== contentEncoding) {
-                            return content;
-                        }
+                    return s3.headObject({
+                            Bucket: bucketName,
+                            Key: blobName
+                        }).promise().then(function(data) {
+                            if (data.ETag !== ('"' + hash + '"') ||
+                                data.CacheControl !== cacheControl ||
+                                data.ContentType !== contentType ||
+                                data.ContentEncoding !== contentEncoding) {
+                                return content;
+                            }
 
-                        // We don't need to upload this file again
-                        skipped++;
-                        return undefined;
-                    })
-                    .catch(function(error) {
-                        errors.push(error);
-                    });
+                            // We don't need to upload this file again
+                            skipped++;
+                            return undefined;
+                        })
+                        .catch(function(error) {
+                            errors.push(error);
+                        });
                 })
                 .then(function(content) {
                     if (!content) {
@@ -443,16 +438,16 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
                         CacheControl : cacheControl
                     };
 
-                    return s3.putObjectAsync(params).then(function() {
-                        uploaded++;
-                    })
-                    .catch(function(error) {
-                        errors.push(error);
-                    });
+                    return s3.putObject(params)
+                        .promise().then(function() {
+                            uploaded++;
+                        })
+                        .catch(function(error) {
+                            errors.push(error);
+                        });
                 });
             }, {concurrency : concurrencyLimit});
-        })
-        .then(function() {
+        }).then(function() {
             console.log('Skipped ' + skipped + ' files and successfully uploaded ' + uploaded + ' files of ' + (totalFiles - skipped) + ' files.');
             if (existingBlobs.length === 0) {
                 return;
@@ -477,22 +472,19 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
                 batches.push(objectsToDelete);
 
                 return Promise.map(batches, function(objects) {
-                    return s3.deleteObjectsAsync({
-                        Bucket : bucketName,
-                        Delete : {
-                            Objects : objects
+                    return s3.deleteObjects({
+                        Bucket: bucketName,
+                        Delete: {
+                            Objects: objects
                         }
-                    })
-                    .then(function() {
+                    }).promise().then(function() {
                         console.log('Cleaned ' + objects.length + ' files.');
                     });
                 }, {concurrency : concurrency});
             }
-        })
-        .catch(function(error) {
+        }).catch(function(error) {
             errors.push(error);
-        })
-        .then(function() {
+        }).then(function() {
             if (errors.length === 0) {
                 done();
                 return;
@@ -504,11 +496,6 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
             });
             done(1);
         });
-    })
-    .catch(function(error) {
-        console.log('Error: Could not load S3 credentials.');
-        done();
-    });
 }
 
 function getMimeType(filename) {
@@ -534,7 +521,7 @@ function getMimeType(filename) {
 
 // get all files currently in bucket asynchronously
 function listAll(s3, bucketName, prefix, files, marker) {
-    return s3.listObjectsAsync({
+    return s3.listObjects.promise({
         Bucket : bucketName,
         MaxKeys : 1000,
         Prefix : prefix,

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -26,7 +26,7 @@ var Promise = require('bluebird');
 var requirejs = require('requirejs');
 var Karma = require('karma').Server;
 var yargs = require('yargs');
-var S3 = require('aws-sdk/clients/s3');
+var AWS = require('aws-sdk');
 var mime = require('mime');
 var compressible = require('compressible');
 
@@ -340,7 +340,7 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
     var gzip = Promise.promisify(zlib.gzip);
     var concurrencyLimit = 2000;
 
-    var s3 = new S3({
+    var s3 = new AWS.S3({
         maxRetries : 10,
         retryDelayOptions : {
             base : 500

